src/ntfsck.c

```diff
@@ -1944,6 +1944,180 @@ static runlist *ntfsck_decompose_runlist(ntfs_attr *na, BOOL *need_fix)
 	return rl;
 }
 
+static int ntfsck_init_root(ntfs_volume *vol, ntfs_inode *ni, ntfs_index_context *ictx)
+{
+	ntfs_attr_search_ctx *ctx = NULL;
+	ntfs_attr *bm_na = NULL;
+	ntfs_attr *ia_na = NULL;
+	INDEX_ROOT *ir = NULL;
+	INDEX_ENTRY *ie = NULL;
+	INDEX_BLOCK *ib = NULL;
+	int ret = STATUS_ERROR;
+	int index_len;
+	int ir_init_size;
+	u32 block_size;
+	s64 r_size;
+	u8 *bm = NULL;
+
+	block_size = ictx->block_size;
+
+	ia_na = ictx->ia_na;
+	if (!ia_na)
+		goto out;
+
+	/* remain one index block not to allocate when add index for meta files in fsck */
+	if (ntfs_attr_truncate(ia_na, block_size))
```

Comment:
we open system files and add index entry for them in ntfsck_initiaiize_root_index(). Is there any reason why one index block should be remained ? because ntfs_index_add_filename will create index block if there is no room in index root ?

Reply:
the reason is to remove possibility of duplicated cluster allocation in fsck.
to remove possibility, remain one $INDEX_ALLOCATION block whose entries length is initialized.
Then ntfs_index_add_filename will not allocate during add meta system files indexes.